services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipInvitationManagerImpl.java

```diff
@@ -76,6 +77,10 @@ public MembershipInvtnSubmission create(UserInfo userInfo,
 		if (!authorizationManager.canAccess(
 				userInfo, mis.getTeamId(), ObjectType.TEAM, ACCESS_TYPE.TEAM_MEMBERSHIP_UPDATE).getAuthorized()) 
 			throw new UnauthorizedException("Cannot create membership invitation.");
+		Team team = teamDAO.get(mis.getTeamId());
+		if (team.getCanPublicJoin() != null && team.getCanPublicJoin()) {
+			throw new IllegalArgumentException("Cannot invite to public team.");
```

Comment:
Also, verify with @jay-hodgson, in case that a team is open for public, how would a user "invite" someone to join beside this method. Is there an alternative way to invite users without creating an MembershipInvitation?